src/domain/relay/limit-addresses.mapper.ts

```diff
@@ -119,48 +119,102 @@ export class LimitAddressesMapper {
   }
 
   private isValidExecTransactionCall(args: { to: string; data: Hex }): boolean {
-    let execTransaction: { data: Hex; to: Hex; value: bigint };
-    // If transaction is an execTransaction
-    try {
-      const safeDecodedData = this.safeDecoder.decodeFunctionData({
-        data: args.data,
+    const execTransactionArgs = this.getExecTransactionArgs(args.data);
+    // Not a valid execTransaction call
+    if (!execTransactionArgs) {
+      return false;
+    }
+
+    // Only ERC-20 transfer to other party is valid
+    if (this.erc20Decoder.helpers.isTransfer(execTransactionArgs.data)) {
+      return this.isValidErc20Transfer({
+        to: args.to,
+        data: execTransactionArgs.data,
       });
-      if (safeDecodedData.functionName !== 'execTransaction') {
-        return false;
-      }
-      const [to, value, data] = safeDecodedData.args;
-      execTransaction = { to, value, data };
-    } catch (e) {
+    }
+
+    // Only ERC-20 transferFrom to other party is valid
+    if (this.erc20Decoder.helpers.isTransferFrom(execTransactionArgs.data)) {
+      return this.isValidErc20TransferFrom({
+        to: args.to,
+        data: execTransactionArgs.data,
+      });
+    }
+
+    // Other ERC-20 methods are valid if they are...
```

Comment:
nitpick: from this point it's not really specific to ERC-20 functions but any function.

Reply:
I removed the comment in edb9749fb6c722676f61148ad36909dfd53d9f79.